task/net/src/server.rs

```diff
@@ -346,19 +449,71 @@ where
         eth: &'a eth::Ethernet,
         mac_address_block: MacAddressBlock,
         bsp: B,
-        storage: &'static mut [Storage; N],
-        sockets: generated::Sockets<'static, N>,
-        mut mkdevice: impl FnMut(usize) -> E,
+        storage: &'static mut [Storage; VLanId::LENGTH],
+        sockets: generated::Sockets<'static, { VLanId::LENGTH }>,
+        mut mkdevice: impl FnMut(VLanId) -> E,
     ) -> Self {
         // Local storage; this will end up owned by the returned ServerImpl.
-        let mut vlan_state: Vec<VLanState<E>, N> = Vec::new();
+        let mut vlan_state: Vec<VLanState<E>, { VLanId::LENGTH }> = Vec::new();
```

Comment:
In the code below, it kinda feels like this would be simplified if this started life as a mutable `EnumMap` instead of getting converted at the end?

Reply:
We talked about this in person; we can't build a default `VLanState`, so we're using `Vec` to incrementally build the array